sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/CatalystTypeConvertersSuite.scala

```diff
@@ -273,6 +274,24 @@ class CatalystTypeConvertersSuite extends SparkFunSuite with SQLHelper {
     assert(errMsg.contains("long overflow"))
   }
 
+  test("SPARK-35726: Truncate java.time.Duration by fields of day-time interval type") {
+    val duration = Duration.ofSeconds(90061)
+    Seq(DayTimeIntervalType(DAY, DAY) -> 86400000000L,
+      DayTimeIntervalType(DAY, HOUR) -> 90000000000L,
+      DayTimeIntervalType(DAY, MINUTE) -> 90060000000L,
+      DayTimeIntervalType(DAY, SECOND) -> 90061000000L,
+      DayTimeIntervalType(HOUR, HOUR) -> 90000000000L,
+      DayTimeIntervalType(HOUR, MINUTE) -> 90060000000L,
+      DayTimeIntervalType(HOUR, SECOND) -> 90061000000L,
+      DayTimeIntervalType(MINUTE, MINUTE) -> 90060000000L,
+      DayTimeIntervalType(MINUTE, SECOND) -> 90061000000L,
+      DayTimeIntervalType(SECOND, SECOND) -> 90061000000L)
+      .foreach { case (dt, value) =>
+        assert(CatalystTypeConverters.createToCatalystConverter(dt)(duration) == value)
```

Comment:
Could you check negative duration, please. Just add this:
```scala
        assert(CatalystTypeConverters.createToCatalystConverter(dt)(duration.negated()) == -value)
```

Reply:
updated

Reply:
the different is right according to the origin logic?
```
 (DayTimeIntervalType(SECOND, SECOND), 90062000000L, -90062000001L))
```